Remove last clang-tidy error with garbage warning

mnurzyns opened the pull request:
There was chance to return undeclarate value due to not enter to
if / else if. There was need for add default return value.

Commit:
Remove last clang-tidy error with garbage warning

There was chance to return undeclarate value due to not enter to
if / else if. There was need for add default return value.

Commit:
Activate clang-tidy on CI

Commit:
Active ALL clang-tisy CI job responsibilities

Commit:
Add c-tor/operator with = default / = delete

For greater understanding class responsibilities = default
and = delete was added. In ChatControl all control classes should
have only one instance and only Controler need to join threads.

Commit:
Remove clang-tidy error after mawoznia changes

Global variables are bad.

marcinwozniak0 commented on MessengerOnFileForLinux/ChatControl/src/ConversationControl.hpp:
```diff
@@ -19,8 +20,15 @@ class ConversationControl
     void conversationEpilog();
+
     ConversationControl(std::shared_ptr<ChatInformation> chatInfo);
+    ConversationControl() = delete;
```
to bym przesunał pod destruktor, odzielił delety enterem. Tak jak jest w Controler.hpp

marcinwozniak0 replied:
+ czy potrzebujemy  ConversationControl() = delete, skoro i tak nie wywołamy domyslengo konstruktora jeśli mamy zdefiniowany jakis z argumentem?

mnurzyns replied:
Masz rację, inaczej próbuje zmatchować (&) lub (&&) z pustym argumentem, wię wystarczy usunięcie tamtych.

marcinwozniak0 commented on MessengerOnFileForLinux/ChatControl/src/TerminalControl.hpp:
```diff
@@ -12,8 +12,15 @@ class TerminalControl
+
     TerminalControl(ChatStatus chatStatus, std::shared_ptr<ChatInformation> chatInfo);
+    TerminalControl() = delete;
     ~TerminalControl() = default;
```
tu tak samo

mnurzyns replied:
k

marcinwozniak0 commented on MessengerOnFileForLinux/ChatControl/src/ConversationControl.hpp:
```diff
@@ -19,8 +20,15 @@ class ConversationControl
     void conversation();
     void conversationEpilog();
+
     ConversationControl(std::shared_ptr<ChatInformation> chatInfo);
```
Dodałbym tutaj explicit

Commit:
Merge branch 'master' of https://github.com/NMarcin/MessengerOnFileForLinux into mnurzyns

Commit:
Remove useless = delete with non-paramet c-tor

ConversationControl and TerminalControl had c-tor with parameter
so it's not needed to delete default c-tor.

Commit:
Remove not needed commits

Error while merging and commit which tried to fix clang-tidy CI issue.
